smart-contract-verifier/smart-contract-verifier/src/lookup_methods/disassemble.rs

```diff
@@ -0,0 +1,67 @@
+use super::opcodes::{opcode, Opcode};
+use bytes::Bytes;
+use std::fmt::{Debug, Display};
+
+pub struct DisassembledOpcode {
+    pub operation: Opcode,
+    pub program_counter: usize,
+    pub args: Vec<u8>,
+}
+
+impl Debug for DisassembledOpcode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.args.is_empty() {
+            writeln!(f, "{}", self.operation.name)
+        } else {
+            writeln!(f, "{} {:?}", self.operation.name, self.args)
+        }
+    }
+}
+
+impl Display for DisassembledOpcode {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        if self.args.is_empty() {
+            write!(f, "{}", self.operation.name)
+        } else {
+            write!(f, "{} {:?}", self.operation.name, self.args)
+        }
+    }
+}
```

Comment:
Nit: just interested: why have the same Debug and Display implementations? Can we just derive Debug implementation? Or at least to implement one through another if they are the same anyway

Reply:
Forgot to clean up this after testing. Removed the Display impl